R/sort_linter.R

```diff
@@ -1,9 +1,24 @@
-#' Require usage of `sort()` over `.[order(.)]`
+#' Check for common mistakes around sorting vectors
+#'
+#' This linter checks for some common mistakes when using [order()] or [sort()].
+#'
+#' First, it requires usage of `sort()` over `.[order(.)]`.
 #'
 #' [sort()] is the dedicated option to sort a list or vector. It is more legible
 #' and around twice as fast as `.[order(.)]`, with the gap in performance
 #' growing with the vector size.
 #'
+#' Second, it requires usage of [is.unsorted()] over equivalents using `sort()`.
+#'
+#' The base function `is.unsorted()` exists to test the sortedness of a vector.
+#'   Prefer it to inefficient and less-readable equivalents like
+#'   `x != sort(x)`. The same goes for checking `x == sort(x)` -- use
+#'   `!is.unsorted(x)` instead.
```

Comment:
Do you think the lint message should reflect the exact replacement, depending on whether it's a regular (`is.unsorted()`) or inverted (`!is.unsorted(x)`) comparison? Or will that be an overkill?

Reply:
The same thought crossed my mind. I think it's a good call b/c sometimes I have to pause a minute and remember which is the equivalent. It's better to write it in the message (and trivial to do so). Thanks!